pypsa/pf.py

```diff
@@ -53,8 +51,7 @@ def imag(X): return np.imag(X.to_numpy())
 def _as_snapshots(network, snapshots):
     if snapshots is None:
         snapshots = network.snapshots
-    if (isinstance(snapshots, six.string_types) or
-        not isinstance(snapshots, (Sequence, pd.Index))):
+    if isinstance(snapshots, str) or not isinstance(snapshots, pd.Index):
         return pd.Index([snapshots])
```

Comment:
You should not drop the test for being list-like completely here! what if you get a list of snapshots?

https://pandas.pydata.org/pandas-docs/version/1.1/reference/api/pandas.api.types.is_list_like.html should be a good way to go